### PR TITLE
add puppeteer example

### PR DESCRIPTION
**Pull request by wnghdcjfe**

### CODE REVIEW

File: puppeteer/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Puppeteer
+
```
Comment (craigtaub) [inline puppeteer/README.md]: Please add small description about Puppeteer.

File: puppeteer/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "example-vue-puppeteer-app",
+  "version": "1.0.0",
+  "description": "Mocha and Puppeteer example",
+  "scripts": {
+    "start": "serve .",
+    "test": "mocha test/*.test.js"
+  },
+  "devDependencies": {
+    "chai": "^4.2.0",
```
Comment (craigtaub) [inline puppeteer/package.json]: Not really needed (see other comments)

File: puppeteer/test/index.test.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    page = await browser.newPage();
+    await page.goto('http://localhost:3000/index.html'); 
+  });
+
+  after(async function() {
+    await browser.close();
+    await server.close();
+  });
+
+  it('should h1 have a Mocha is good', async function() {
```
Comment (craigtaub) [inline puppeteer/test/index.test.js]: Confusing spec. Perhaps `h1 should say "mocha is good"`

File: puppeteer/package.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "example-vue-puppeteer-app",
+  "version": "1.0.0",
+  "description": "Mocha and Puppeteer example",
+  "scripts": {
+    "start": "serve .",
+    "test": "mocha test/*.test.js"
```
Comment (craigtaub) [inline puppeteer/package.json]: Please rename file.to `spec.js` and then can just run `mocha`. I think it can confuse people why it's needed to have this.

**Comment by craigtaub** [comment]
Please add link to main README

File: puppeteer/README.md [hunk 5]
```diff
@@ -0,0 +1,6 @@
...
+## Commands
+
+- `npm start` - Start a static server using [serve](https://www.npmjs.com/package/serve). Run `npm start` and go to `http://localhost:5000` to see the basic Vue app in action.
```
[the remark this answers is not in the document]
Comment (wnghdcjfe) [reply puppeteer/README.md]: thanks a lot.

File: puppeteer/package.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "example-vue-puppeteer-app",
```
[the remark this answers is not in the document]
Comment (wnghdcjfe) [reply puppeteer/package.json]: thanks a lot.

File: puppeteer/package.json (hunk 4 above)
Comment (wnghdcjfe) [reply puppeteer/package.json]: `"test": "mocha test/*.spec.js"` like this? and `index.test.js` file rename  `index.spec.js` ?

Comment (craigtaub) [reply puppeteer/package.json]: As in `"test" : "mocha"`. Its better practice to omit options on CLI unless they are really needed.

**Comment by craigtaub** [comment]
Great stuff 👍 . Thanks very much for the contribution 😄

**Pull request merged**